services/director-v2/src/simcore_service_director_v2/modules/dynamic_sidecar/scheduler/_core/_scheduler.py

```diff
@@ -464,13 +464,14 @@ async def retrieve_service_inputs(
         )
         duration = time.time() - started
 
-        get_instrumentation(
-            self.app
-        ).dynamic_sidecar_metrics.input_ports_pull_rate.labels(
-            **get_metrics_labels(scheduler_data)
-        ).observe(
-            get_rate(transferred_bytes, duration)
-        )
+        if rate := get_rate(transferred_bytes, duration) >= 1:
+            get_instrumentation(
+                self.app
+            ).dynamic_sidecar_metrics.input_ports_pull_rate.labels(
+                **get_metrics_labels(scheduler_data)
+            ).observe(
+                rate
```

Comment:
Q: do you get to call `retrieve_service_inputs` many times within the same session? i.e. I mean, do you need ot have a mechanism to "change" the instrumentation to new rate? of this call can do that?

Reply:
I'd say this call is being issued multiple times per session. And it's just a counter that gets set. This is then scraped by prometheus.